Library/Homebrew/global.rb

```diff
@@ -30,7 +30,7 @@
 HOMEBREW_USER_AGENT_CURL = ENV["HOMEBREW_USER_AGENT_CURL"]
 HOMEBREW_USER_AGENT_RUBY = "#{ENV["HOMEBREW_USER_AGENT"]} ruby/#{RUBY_VERSION}-p#{RUBY_PATCHLEVEL}"
 
-HOMEBREW_CURL_ARGS = "-f#RLA"
+HOMEBREW_CURL_ARGS = "--fail --progress-bar --remote-time --location --user-agent"
```

Comment:
What do you think about making this an array instead? This will also make proper argument assembly where this is used easier, and filtering individual `curl` arguments won't need to rely on `String#sub` and could instead use `Array#-` (e.g. `HOMEBREW_CURL_ARGS - %w[--progress-bar]`). It seems like adding the user agent at this stage would make sense, too. (But I'm less sure about this second part.) To be more specific:

```ruby
HOMEBREW_CURL_ARGS = %W[
  --fail
  --progress-bar
  --remote-time
  --location
  --user-agent #{HOMEBREW_USER_AGENT_CURL}
]
```

(Code using this will of course need to be adjusted slightly, but mostly just can omit the `.split`.)